From 225061c30710c4e612a416a355f231a01e03e64e Mon Sep 17 00:00:00 2001
From: Harsh J <harsh@apache.org>
Date: Sat, 1 Sep 2012 19:22:11 +0000
Subject: [PATCH 1712/2748] HDFS-2580. NameNode#main(...) can make use of
 GenericOptionsParser. Contributed by harsh.
 (harsh)

(cherry picked from commit 54e612bfb9f877e58f7f153c43cb4147876826d3)
(cherry picked from commit 916f949131afa2d6f20bf654f5fde40377c5f618)

Change-Id: I91892d5baa7187475800dfb01325fb8510fe8758
---
 .../hadoop/hdfs/server/namenode/NameNode.java      |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNode.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNode.java
index cbc7ead..05d54d8 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNode.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNode.java
@@ -70,6 +70,7 @@
 import org.apache.hadoop.tracing.TraceUtils;
 import org.apache.hadoop.tracing.TracerConfigurationManager;
 import org.apache.hadoop.util.ExitUtil.ExitException;
+import org.apache.hadoop.util.GenericOptionsParser;
 import org.apache.hadoop.util.JvmPauseMonitor;
 import org.apache.hadoop.util.ServicePlugin;
 import org.apache.hadoop.util.StringUtils;
@@ -1472,6 +1473,10 @@ public static NameNode createNameNode(String argv[], Configuration conf)
     LOG.info("createNameNode " + Arrays.asList(argv));
     if (conf == null)
       conf = new HdfsConfiguration();
+    // Parse out some generic args into Configuration.
+    GenericOptionsParser hParser = new GenericOptionsParser(conf, argv);
+    argv = hParser.getRemainingArgs();
+    // Parse the rest, NN specific args.
     StartupOption startOpt = parseArguments(argv);
     if (startOpt == null) {
       printUsage(System.err);
-- 
1.7.9.5

